### ISSUE TITLE
Option to specify an anchor point for zoom interactions

### ISSUE THREAD

**Issue by tiecia** [openlayers/openlayers#16004]
**Is your feature request related to a problem? Please describe.**
There are times where I need to zoom around a specific anchor point that is not the center or mouse position when using either the KeyboardZoom, MouseWheelZoom, or PinchZoom interactions. I do not believe this can easily be done with the current implementations of these interactions.

**Describe the solution you'd like**
I am suggesting an option such as `anchorCoordinate` or `anchorPixel` in each zoom interaction, or at least the ones that zoom around a single point on the map, that when set cause the interaction to zoom around that coordinate rather then the center of the map or mouse cursor.

If this is already possible please let me know but I have searched the docs and cannot find a way to do this.

**Comment by ahocevar** [comment]
Maybe not a 100% fit for what you want, but what I use on similar use cases is the `padding` option of `ol/View.js`.

**Comment by tiecia** [comment]
Reading the description of `padding` in the docs it seems like that would do what I need. I haven't tried it for my needs yet so don't know if would cause any side effects.

I ended up modifying the interaction classes for my own project with an added `anchorCoordinate` option and can submit a PR if there is interest.

**Comment by grahamcracker1234** [comment]
I would be interested to see how you accomplished this, as a feature like this would be very useful.

**Comment by tiecia** [comment]
You can check out my implementation on my fork: https://github.com/tiecia/openlayers/tree/zoom-interaction-anchor

I also created an example showing how to use it:
https://github.com/tiecia/openlayers/blob/zoom-interaction-anchor/examples/zoomAnchor.js

This implementation was been working well for what I need.

**Comment by grahamcracker1234** [comment]
This is great! Exactly what I was looking for, thank you. Frankly quite surprised this is not built in directly to OpenLayers

**Comment by marcjansen** [comment]
@ahocevar do you think this would be good in the library? I think it is a useful addition.

**Comment by ahocevar** [comment]
@marcjansen I think the proposed API is a bit cumbersome to use. If we want something like that, the user friendly way would be to have a single `setZoomAnchor()` method on the `View`. It could then be set internally as a `ViewState` property, and used by all interactions that deal with anchored zooming.

**Comment by grahamcracker1234** [comment]
I agree with @ahocevar, if this functionality were to be built directly into OpenLayers, an additional method on the `View` like `setZoomAnchor()` would simplify the use-cases for a vast majority of users who want this functionality. But I also think the initial `setAnchorCoordinate()` functionality should stay intact for those who want more advanced control.
